### PR TITLE
Fixed compatibility of battery compartment mod with certain items

### LINKED ISSUE #25041: Remote vehicle controller does not work with battery compartment mod
**Describe the bug**  
Remote vehicle controller reports empty batteries when a battery compartment mod is installed with a charged battery.

**To Reproduce**  
1. Install a battery compartment mod to a remote vehicle controller.
2. Reload with a fully charged car battery.
3. Try to control a remote-controllable vehicle.
4. The remote vehicle controller will consume one charge and then tell that the batteries are out and abort control.

**Expected behavior**  
It should control the vehicle continuously, just like it does with conventional batteries.

**Versions and configuration(please complete the following information):**
 - Windows 10 x64
 - Game Version 0.C-30764-g1bb4be6
 - Graphics version: Tiles
 - Mods loaded `[
  "dda",
  "Experimental-Z-Level",
  "More_Survival_Tools",
  "blazemod",
  "Tanks",
  "Urban_Development",
  "boats",
  "filthy_morale",
  "makeshift",
  "more_buildings",
  "more_locations",
  "mutant_npcs",
  "national_guard_camp",
  "npc_traits",
  "craftgp",
  "brightnights",
  "bio_mod"
]`

### LINKED ISSUE #26670: multi-cooker with tool mod installed considered filled with dish
**Describe the bug**  
multi-cooker with tool mod will promt for dish removal when used. In that case the mod itself will be removed.

**To Reproduce**  
1. Obtain multi-cooker.
2. Install any tool mod to it.
3. Activate multi-cooker.

**Expected behavior**  
Multi-cooker can be used to cook food with any tool mod installed. Tool mods are impossible to remove.

**Screenshots**  
No

**Versions and configuration(please complete the following information):**
 - OS: Windows 7
 - Game Version: 0.C-34483-ga08bd8c
 - Graphics version: Tiles
 - Mods loaded: 
  "dda",
  "no_npc_food",
  "filthy_morale",
  "novitamins",
  "realguns",
  "FIC_Weapons",
  "More_Survival_Tools",
  "Medieval_Stuff",
  "makeshift",
  "craftgp",
  "safeautodoc",
  "brightnights",
  "mutant_npcs",
  "Mining_Mod",
  "alt_map_key",
  "npc_traits",
  "more_locations",
  "FujiStruct",
  "cityside",
  "oa_additional_buildings_mod",
  "Urban_Development",
  "boats",
  "blazemod",
  "deoxymod",
  "Tanks",
  "EngineeringEssentials",
  "StatsThroughSkillsFour",
  "more_classes_scenarios",
  "manualbionicinstall",
  "DP_AMAZING_CHEST_INDICATORS",
  "Cata++"

### PR DESCRIPTION
**Pull request by DarkDaskin**
#### Summary
SUMMARY: Bugfixes "Fixed compatibility of battery compartment mod with certain items"

#### Purpose of change
Fixes #25041, fixes #26670 and some other items which do not work correctly with battery compartment mod installed:

- remote vehicle controller
- RC controller
- RC car
- multi-cooker

#### Describe the solution
Code now uses ammo functions rather than accesses `charges` directly. Also added filtering to code which assumed specific item contents.

#### Describe alternatives you've considered
Disallow installing mods to affected items.

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
Could you please also test if your changes fix #20384?

File: src/map.cpp [hunk 1]
```diff
@@ -4919,17 +4919,22 @@ static bool trigger_radio_item( item_stack &items, std::list<item>::iterator &n,
...
-        // If that changes, this needs logic to handle the alternative.
-        n->convert( n->contents.front().typeId() );
-        if( n->has_flag( "RADIO_INVOKE_PROC" ) ) {
-            n->process( nullptr, pos, true );
-        }
+    } else if( n->has_flag( "RADIO_CONTAINER" ) && !n->contents.empty() ) {
+        auto it = std::find_if( n->contents.begin(), n->contents.end(), [ &signal ]( const item &c ) {
+            return c.has_flag( signal );
+        } );
+        if ( it != n->contents.end() ) {
```
Comment (ZhilkinSerg) [inline src/map.cpp]:
```suggestion
        if( it != n->contents.end() ) {
```

File: src/iuse.cpp [hunk 2]
```diff
@@ -6541,18 +6544,8 @@ int iuse::radiocar( player *p, item *it, bool, const tripoint & )
...
-            bomb = it->contents.front();
-        }
-
         it->convert( "radio_car_on" ).active = true;
-
-        if( !( bomb.is_null() ) ) {
-            it->put_in( bomb );
-        }
-
+
```
Comment (ZhilkinSerg) [inline src/iuse.cpp]:
```suggestion

```

File: src/iuse.cpp [hunk 3]
```diff
@@ -6619,18 +6611,8 @@ int iuse::radiocaron( player *p, item *it, bool t, const tripoint &pos )
...
-            bomb = it->contents.front();
-        }
-
         it->convert( "radio_car" ).active = false;
-
-        if( !( bomb.is_null() ) ) {
-            it->put_in( bomb );
-        }
-
+
```
Comment (ZhilkinSerg) [inline src/iuse.cpp]:
```suggestion

```

File: src/iuse.cpp [hunk 4]
```diff
@@ -7045,11 +7028,16 @@ int iuse::multicooker( player *p, item *it, bool t, const tripoint &pos )
         uilist menu;
         menu.text = _( "Welcome to the RobotChef3000.  Choose option:" );
 
+        // Find actual contents rather than attached mod or battery.
+        auto dish_it = std::find_if_not(it->contents.begin(), it->contents.end(), [] ( const item &c ) {
```
Comment (ZhilkinSerg) [inline src/iuse.cpp]:
```suggestion
        auto dish_it = std::find_if_not( it->contents.begin(), it->contents.end(), [] ( const item &c ) {
```

**Comment by DarkDaskin** [comment]
@Night-Pryanik the bomb now explodes, but the error message stays.

**Pull request merged**